File: javascript/test/ParameterTypeTest.ts
import * as assert from 'node:assert'

import ParameterType from '../src/ParameterType.js'
import ParameterTypeRegistry from '../src/ParameterTypeRegistry.js'

describe('ParameterType', () => {
  it('does not allow ignore flag on regexp', () => {
    assert.throws(
      () => new ParameterType('case-insensitive', /[a-z]+/i, String, (s) => s, true, true),
      { message: "ParameterType Regexps can't use flag 'i'" }
    )
  })

  it('has a type name for {int}', () => {
    const r = new ParameterTypeRegistry()
    const t = r.lookupByTypeName('int')!
    // @ts-expect-error
    assert.strictEqual(t.type.name, 'Number')
  })

  it('has a type name for {bigint}', () => {
    const r = new ParameterTypeRegistry()
    const t = r.lookupByTypeName('biginteger')!
    // @ts-expect-error
    assert.strictEqual(t.type.name, 'BigInt')
  })

  it('has a type name for {word}', () => {
    const r = new ParameterTypeRegistry()
    const t = r.lookupByTypeName('word')!
    // @ts-expect-error
    assert.strictEqual(t.type.name, 'String')
  })
})

File: dart/test/parameter_type_test.dart
import 'package:cucumber_expressions/src/errors.dart';
import 'package:cucumber_expressions/src/parameter_type.dart';
import 'package:cucumber_expressions/src/parameter_type_lookup.dart';
import 'package:cucumber_expressions/src/parameter_type_registry.dart';
import 'package:test/test.dart';

void main() {
  group('ParameterType', () {
    test('does not allow ignore flag on regexp', () {
      expect(
        () => ParameterType<String?>(
          'case-insensitive',
          RegExp('[a-z]+', caseSensitive: false),
          'String',
          (s) => s.first,
          useForSnippets: true,
          preferForRegexpMatch: true,
        ),
        throwsA(
          isA<CucumberExpressionException>().having(
            (e) => e.message,
            'message',
            equals("ParameterType Regexps can't use flag 'i'"),
          ),
        ),
      );
    });

    test('has a type name for {int}', () {
      final r = ParameterTypeRegistry();
      final t = lookupParameterTypeByName(r, 'int')!;
      expect(t.type, equals('int'));
    });

    test('has a type name for {biginteger}', () {
      final r = ParameterTypeRegistry();
      final t = lookupParameterTypeByName(r, 'biginteger')!;
      expect(t.type, equals('BigInt'));
    });

    test('has a type name for {word}', () {
      final r = ParameterTypeRegistry();
      final t = lookupParameterTypeByName(r, 'word')!;
      expect(t.type, equals('String'));
    });
  });
}
